### PR TITLE
Migrate remaining shotguns and update ammo [RDY]

### LINKED ISSUE #17359: Conversion kits cannot be used on shotguns
The descriptive text for the .223 caliber conversion kit and the .308 caliber conversion kit says they can be used to convert a shotgun into a 223 or 308 rifle, but they lack a shotgun entry in their mod_targets field.

### PR DESCRIPTION
**Pull request by mugling**
Guns:
- Migrates Keltech KSG and forbid adding `stock` modifications
- Adds appropriate `reload_noise` for `shotgun_pump` (inspired by #17366)
- Rebalances Leadworks shotgun to have similar theme to other Leadworks items (02a1014)
- Ensure all gun barrels are made of `steel` and not `iron`
- Increase volume of saiga to account for it's `barrel_length`

Ammo:
- Ensure crafting recipes appropriate for stack size (especially number of primers/hulls etc)
- Similarly adjust deconstruction recipes to match above
- Crafting requires power only if forming shot/slugs from lead (not for scrap shots or flechette)
- Crafting shot requires `lead` (with the exception of scrap shot or flechette)
- Nails nolonger substitute for `combatnail` which is now known as `flechette`
- Scrap shot has very low requirements to offset it's poor performance

Mods:
- Allow conversion of shotguns to `308` and `223` ammo (fixes #17359)

### CODE REVIEW

**Comment by BorkBorkGoesTheCode** [comment]
The use of iron in the antique double-action shotguns may be justified.

**Comment by mugling** [comment]
That would be very old indeed

File: data/json/items/gunmod/bore.json [hunk 1]
```diff
@@ -33,8 +33,8 @@
     "symbol": ":",
     "color": "green",
     "location": "bore",
-    "mod_targets": [ "rifle" ],
```
Comment (BorkBorkGoesTheCode) [inline data/json/items/gunmod/bore.json]: It may be more realistic to create new conversion kits instead of extending these.

**Comment by mugling** [comment]
Why different conversion kits, what is different about them given the answer in #17359 that they replace the barrel?

**Comment by mugling** [comment]
Have resolved merge conflicts

**Pull request merged**